common-npm-packages/kubernetes-common/kubelogin.ts

```diff
@@ -63,9 +63,7 @@ export class Kubelogin {
 
       if (authType == 'spnCertificate') {
         taskLib.debug('certificate based endpoint');
-        let certificateContent: string = taskLib.getEndpointAuthorizationParameter(connectedService, 'servicePrincipalCertificate', false);
-        servicePrincipalKey = path.join(taskLib.getVariable('Agent.TempDirectory') || taskLib.getVariable('system.DefaultWorkingDirectory'), 'spnCert.pem');
-        fs.writeFileSync(servicePrincipalKey, certificateContent);
+        throw taskLib.loc('AuthSchemeNotSupported', `${authScheme} with a certificate`);

```

Comment:
**Suggestion:**
Maybe it is possible to convert pem to pfx with `node-forge` or similar library

Reply:
Yes, it worked. I added the conversion from pem to pfx. Thank you!